### PR TITLE
Feature/46 get feed list

### LINKED ISSUE #46: [Feat] 피드 목록 조회 API
## 📌  Description
- 사용자는 커뮤니티 뷰에서 피드 목록을 조회할 수 있다.
## 📝  To-do
- [x] 피드 목록 조회 API 구현

### PR DESCRIPTION
**Pull request by pingowl**
## 📌 관련 이슈
- close #46
## ✨ 작업 내용
- 피드 목록 조회 API 구현
- 레코드에 isWrriten 필드(특정 레코드에 대한 피드 작성여부) 추가

### CODE REVIEW

File: src/main/java/beachcombine/backend/service/FeedService.java [hunk 1]
```diff
@@ -23,17 +28,21 @@ public class FeedService {
...
     public Long saveFeed(Long memberId, FeedSaveRequest request, Long recordId) throws IOException {
 
         Record findRecord = getRecordOrThrow(recordId);
+        if(findRecord.getFeed() != null ){
+            throw new CustomException(ErrorCode.EXIST_FEED_RECORD);
+        }
 
         Feed feed = Feed.builder()
                 .review(request.getReview())
+                .record(findRecord)
```
Comment (ChooSeoyeon) [inline src/main/java/beachcombine/backend/service/FeedService.java]: 43라인은 아래 45라인과 중복되는 코드라서 없어도 될 거 같습니당

File: src/main/java/beachcombine/backend/service/FeedService.java [hunk 2]
```diff
@@ -23,17 +28,21 @@ public class FeedService {
...
     private final RecordRepository recordRepository;
+    private final ImageService imageService;
 
     // 피드 기록하기
     public Long saveFeed(Long memberId, FeedSaveRequest request, Long recordId) throws IOException {
 
         Record findRecord = getRecordOrThrow(recordId);
+        if(findRecord.getFeed() != null ){
+            throw new CustomException(ErrorCode.EXIST_FEED_RECORD);
+        }
```
Comment (ChooSeoyeon) [inline src/main/java/beachcombine/backend/service/FeedService.java:39]: 중복 검사 넣은 거 좋습니다

File: src/main/java/beachcombine/backend/service/RecordService.java [hunk 3]
```diff
@@ -65,6 +65,7 @@ public List<RecordResponse> getRecordList(Long memberId) {
         List<Record> recordList = recordRepository.findAllByMemberId(memberId);
         for (Record record: recordList){
 
+            Boolean isWritten = (record.getFeed() != null);
```
Comment (ChooSeoyeon) [inline src/main/java/beachcombine/backend/service/RecordService.java:68]: 제가 개인적으로 궁금했던 부분인데, 이 방법과 feed에서 recordId로 검색하는 방법 중 어떤 게 더 좋을까요? 정확성, 성능 측면에서요! 쿼리 날라가는 거 확인해보면 좋을 거 같아용!

Comment (pingowl) [reply src/main/java/beachcombine/backend/service/RecordService.java:68]: header Authorization : Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9.eyJzdWIiOiIzMyIsInJvbGUiOiJST0xFX1VTRVIiLCJpZCI6MSwiZXhwIjoxNjc4MTE4ODU2LCJ1c2VybmFtZSI6IjMzIn0.9IRex2Je2Voyo4j08fpsvbXeMUZ_OMQw8f0kn8AvtThflltHCaq46RdJ62L7_IsW44jPQVRc03Z8O3vWLMeFmg

Hibernate: **select** member0_.member_id as member_i1_2_, member0_.created_date as created_2_2_, member0_.modified_date as modified3_2_, member0_.email as email4_2_, member0_.image as image5_2_, member0_.login_id as login_id6_2_, member0_.month_point as month_po7_2_, member0_.nickname as nickname8_2_, member0_.password as password9_2_, member0_.profile_public as profile10_2_, member0_.provider as provide11_2_, member0_.purchase_point as purchas12_2_, member0_.role as role13_2_, member0_.total_point as total_p14_2_ **from** member member0_ where member0_.login_id=?

Hibernate: **select** record0_.record_id as record_i1_3_, record0_.created_date as created_2_3_, record0_.modified_date as modified3_3_, record0_.after_image as after_im4_3_, record0_.beach_id as beach_id8_3_, record0_.before_image as before_i5_3_, record0_.distance as distance6_3_, record0_.duration as duration7_3_, record0_.feed_id as feed_id9_3_, record0_.member_id as member_10_3_ from record record0_ **left outer join** member member1_ on record0_.member_id=member1_.member_id where member1_.member_id=?

Comment (ChooSeoyeon) [reply src/main/java/beachcombine/backend/service/RecordService.java:68]: 테스트용

`Boolean isWritten = feedRepository.findByRecordId(record.getId());`

FeedRepository에다가는
`boolean findByRecordId(Long recordId);`

Comment (pingowl) [reply src/main/java/beachcombine/backend/service/RecordService.java:68]: header Authorization : Bearer 

Hibernate: select member0_.member_id as member_i1_2_, member0_.created_date as created_2_2_, member0_.modified_date as modified3_2_, member0_.email as email4_2_, member0_.image as image5_2_, member0_.login_id as login_id6_2_, member0_.month_point as month_po7_2_, member0_.nickname as nickname8_2_, member0_.password as password9_2_, member0_.profile_public as profile10_2_, member0_.provider as provide11_2_, member0_.purchase_point as purchas12_2_, member0_.role as role13_2_, member0_.total_point as total_p14_2_ from member member0_ where member0_.login_id=?

Hibernate: select record0_.record_id as record_i1_3_, record0_.created_date as created_2_3_, record0_.modified_date as modified3_3_, record0_.after_image as after_im4_3_, record0_.beach_id as beach_id8_3_, record0_.before_image as before_i5_3_, record0_.distance as distance6_3_, record0_.duration as duration7_3_, record0_.feed_id as feed_id9_3_, record0_.member_id as member_10_3_ from record record0_ left outer join member member1_ on record0_.member_id=member1_.member_id where member1_.member_id=?

**Review by ChooSeoyeon — APPROVED** [review]
전부 확인했습니다. 고생하셨습니다!

**Pull request merged**